app/Libraries/UKCP.php

```diff
@@ -140,4 +144,38 @@ public static function getPathForToken($tokenID, $account)
     {
         return self::TOKEN_PATH_ROOT.$account->id.'/'.$tokenID.'.json';
     }
+
+    public function getStandStatus(string $airfield): array
+    {
+        try {
+            $cached = Cache::get($this->getStandStatusCacheKey($airfield));
+            if ($cached) {
+                return $cached;
+            }
+
+            $response = $this->client->get(
+                sprintf('%s/api/stand/status?airfield=%s', config('services.ukcp.url'), $airfield),
+                ['timeout' => 5]
+            );
+            $body = json_decode($response->getBody()->getContents(), true);
+
+            return tap(
+                collect($body['stands'])->sortBy('identifier', SORT_NUMERIC)->values()->toArray(),
+                fn (array $stands) => Cache::put(
+                    $this->getStandStatusCacheKey($airfield),
+                    collect($stands)->sortBy('identifier', SORT_NUMERIC)->values()->toArray(),
+                    Carbon::parse($body['refresh_at'])
+                )
+            );
```

Comment:
Looks like a candidate for a bit of `Cache::remember()` https://laravel.com/docs/10.x/cache#retrieve-store

Reply:
Unfortunately `remember` doesn't work in this instance, as we'd need to know the `ttl` up-front, whereas the `ttl` is determined by the response we get :)

Reply:
I did a bit of digging, and `remember` does accept `$ttl` to be a Closure of form `fn ($cachedValue) => int`, so you potentially could refactor so that the refresh_at is passed back, but appreciate that might be a bit OTT!